src/vs/platform/quickinput/browser/quickAccess.ts

```diff
@@ -189,10 +190,10 @@ export class QuickAccessController extends Disposable implements IQuickAccessCon
 
 		// Whenever the value changes, check if the provider has
 		// changed and if so - re-create the picker from the beginning
-		disposables.add(picker.onDidChangeValue(value => {
-			const [providerForValue] = this.getOrInstantiateProvider(value);
+		disposables.add(picker.onDidChangeValue(async value => {
+			const [providerForValue] = await this.getOrInstantiateProvider(value);
```

Comment:
If this listener becomes `async` we probably need some logic to cancel a pending handler when the event fires again to ensure we always only execute on the latest.